controller/destination/listener.go

```diff
@@ -76,11 +86,8 @@ func (l *endpointListener) toWeightedAddrSet(endpoints []common.TcpAddress) *pb.
 				log.Errorf("Could not find pod for IP [%s], this IP will be sent with no metric labels.", ipAsString)
 			} else {
 				pod := resultingPods[0]
-				metricLabelsForPod = map[string]string{
-					"pod": pod.Name,
-				}
-
-				namespace = pod.Namespace
```

Comment:
as we were previously getting `namespace` from the `pod`, what do you think about adding a check here validating that each `l.labels["service"] == pod.Namespace` ?

Reply:
Do you mean `l.labels["namespace"] == pod.Namespace`? We could add this check, but I'm not sure what we'd do in the case that the check fails. The previous implementation returned the namespace of the last addr in the set, but it never validated that all addrs were in the same namespace. The new implementation sets the namespace based off the service id itself, which I think is more correct, so I'm inclined to just leave this.

Reply:
Yeah, I meant `l.labels["namespace"]`. It did seem strange we were relying on the namespace of the last addr. I suppose if the check ever failed, there's something bigger and unexpected happening. Cool to leave this as is.